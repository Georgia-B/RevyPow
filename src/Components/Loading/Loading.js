import React from 'react';

const Loading = () => {
    return (
        <div className="loading">
            <div className="loading__spinner">
                <div className="loading__spinner__item"></div>
                <div className="loading__spinner__item"></div>
                <div className="loading__spinner__item"></div>
                <div className="loading__spinner__item"></div>
            </div>
        </div>
    );
}

export default Loading;
