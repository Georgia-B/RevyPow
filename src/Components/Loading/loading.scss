.loading {
    width: 80%;
    margin: auto;
    text-align: center;
}

.loading__spinner {
    display: flex;
    position: relative;
    width: 64px;
    height: 64px;
    margin: 30px auto;
}

.loading__spinner__item {
    box-sizing: border-box;
    display: block;
    position: absolute;
    width: 51px;
    height: 51px;
    margin: 6px;
    border: 4px solid #ffffff;
    border-radius: 50%;
    animation: lds-ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
    border-color: #ffffff transparent transparent transparent;
}

.loading__spinner__item:nth-child(1) {
    animation-delay: -0.45s;
}

.loading__spinner__item:nth-child(2) {
    animation-delay: -0.3s;
}

.loading__spinner__item:nth-child(3) {
    animation-delay: -0.15s;
}


@keyframes lds-ring {
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
}