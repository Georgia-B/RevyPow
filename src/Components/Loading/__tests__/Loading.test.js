import React from 'react'
import { cleanup, render } from '@testing-library/react'
import Loading from '../Loading';

describe('Components | <Loading />', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<Loading />);
        expect(component.asFragment()).toMatchSnapshot();
    })
})