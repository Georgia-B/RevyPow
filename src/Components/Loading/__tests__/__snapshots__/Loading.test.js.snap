// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | <Loading /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="loading"
  >
    <div
      class="loading__spinner"
    >
      <div
        class="loading__spinner__item"
      />
      <div
        class="loading__spinner__item"
      />
      <div
        class="loading__spinner__item"
      />
      <div
        class="loading__spinner__item"
      />
    </div>
  </div>
</DocumentFragment>
`;
